test/cpp/tensorexpr/test_boundsinference.cpp

```diff
@@ -749,6 +749,29 @@ TEST(BoundsInference, HasPartialOverlapSameBufferWithOverlap) {
   ASSERT_TRUE(hasPartialOverlap(analyzer, forK, forJ));
 }
 
+TEST(BoundsInference, HasPartialOverlapSameBufferWithFullOverlap) {
+  KernelScope kernel_scope;
+
+  // Input IR:
+  //   for (int j = 10; j < 100; j++) {
+  //     A[j] = 10 * j;
+  //   }
+  //   for (int k = 10; k < 100; k++) {
+  //     A[k] = 20 * k;
+  //   }
+  BufHandle a_buf("A", {200}, kInt);
+  VarHandle j("j", kInt);
+  VarHandle k("k", kInt);
+  auto forJ = For::make(j, 10, 100, Store::make(a_buf, {j}, Mul::make(10, j)));
+  auto forK = For::make(k, 10, 100, Store::make(a_buf, {k}, Mul::make(20, k)));
+  auto par = Block::make({forJ, forK});
+
+  tensorexpr::analysis::MemDependencyChecker analyzer;
+  par->accept(&analyzer);
+  ASSERT_FALSE(hasPartialOverlap(analyzer, forJ, forK));
+  ASSERT_FALSE(hasPartialOverlap(analyzer, forK, forJ));
```

Comment:
Why is it `false`? Aren't both loops writing to `A[10:100]`?

Reply:
This got me thinking more about the overlaps. In general using bounds information does not give details about the accesses every iteration, it only gives a summary of the full access in the loop. So, it is better to be conservative here and handle any overlap as conflicting. This is going to miss some of the cases where we can fuse. But thats better than possibly being incorrect. 

I will work on figuring a better way to catch other missing fusion opportunities.